src/Core/Validation/FieldValidation/TimeFieldValidator.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\FieldValidation\DateFieldValidator;
+
+/**
+ * Validates that a value is a valid time, which means that it follows the equivalent formats:
+ * - PHP date format H:i:s
+ * - ISO format 'HH:mm:ss' i.e. DBTime::ISO_TIME
+ */
+class TimeFieldValidator extends DateFieldValidator
```

Comment:
We have seperate DBDate, DBDatetime and DBTime fields, likewise we have separate DateField, DatetimeField and TimeField form fields

I think it's very intuitive to also have matching DateFieldValidator, DatetimeFieldValidator and TimeFieldValidators too, so I'll keep these 3x FieldValidators

I think the DateFieldValidator make sense as a subclass, simply for the the fact we use the native PHP date_parse_from_format() to do all of the work ... means that "a time" kinds of is "a date". If you really wanted an abstract base class, what would you call it? I can't think of anything here that doesn't include the word "date" in it

Reply:
I don't think it's intuitive to have the three validators - just because there's three separate `DBField` classes does _not_ mean we have separate validators. The validators _do not need_ to map to `DBField` implementations. They're intentionally abstracted from `DBField`, and won't exclusively be used with `DBField` classes.

> I think the DateFieldValidator make sense as a subclass, simply for the the fact we use the native PHP date_parse_from_format() to do all of the work

The function we use in the implementation of a validator should not dictate what the name of a class or the class hierarchy is. Time is not a date. 

> If you really wanted an abstract base class, what would you call it?

What I really want is to just have the one class to handle all of these, as there seems to be no good reason to have them separate.
If they _must_ be separate (and "there's separate DBField classes" is not a good enough reason for me to accept that), then something like "TemporalValidator" might work. If they're a single class, `DateTimeValidator` seems appropriate.

Reply:
On second thought I can't see you ever agreeing with me here so just leave it